### DISCUSSION TITLE
Crossref over multiple files

### DISCUSSION THREAD

**Discussion by mingsu** [quarto-dev/quarto-cli#2900] [Q&A] [answered]
Crossref over multiple files is highly expected.

Considering such a use case, the `Main text` refer to a figure given by `Supplementary material`.

Is it possible to accomplish this? Below is a MWE.

`cat main.qmd`

````qmd
---
title: main text
---


see [@fig-aaa].



````

`cat supplementary.qmd`

````qmd
---
title: supplementary materia
---


see [@fig-aaa].

```{r}
#| label: fig-aaa
#| fig-cap: aaa
plot(1)
```


````

**Comment by cscheid** [comment]
I believe this works for book projects, but not for website projects. I agree that it would be ideal to support it more generally, but we don't have that right now.

**Reply by emmansh** [reply]
What option do I have right now to refer (via hyperlink) from one page to a different web page in my quarto website?

**Reply by dragonstyle** [reply]
You can link to `qmd` files (including hashes) and the rendered content will use the correct output file extension. 

`See [section on topic](topics.qmd#section)`

Should work.

**Reply by emmansh** [reply]
Thanks. But what if I have different `.qmd`s with the same name (but in different folders)? In my quarto website, the convention is that each post is named `index.qmd` and lives in its own directory. I want to link between posts, or generally link between any two files in my website, that may or may not have a unique name.

```shell
|   index.qmd
|   posts.qmd
|   _quarto.yml
|   
+---.quarto
+---docs
\---posts
    +---blog_post_1
    |       index.qmd      # <--- I want to hyperlink between here 
    |                              
    \---blog_post_2      
            index.qmd      # <--- and here
```

**Reply by cscheid** [reply, marked as answer]
Use one of:

- relative paths: `[link to first blog post](../blog_post_1/index.qmd)`. Relative paths are always resolved with respect to the document they reside in.
- "absolute" paths: `[link to first blog post](/blog_post_1/index.qmd)`. Absolute paths are always resolved with respect to the project root directory.

**Reply by emmansh** [reply]
Thanks! Works like a charm :)
